### PR TITLE
Start to organize unit tests

### PR DESCRIPTION
**Pull request by rocky**
Make the directory structure more closely resemble how it appears in the module space.

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/247"></a>

### CODE REVIEW

File: test/test_convert.py [hunk 1]
```diff
@@ -1,191 +0,0 @@
-#!/usr/bin/env python3
```
Comment (rocky) [inline test/test_convert.py:1]: I am not sure why this show up as deleted since it was _moved_ to `test/parser/test_convert.py` above.

**Comment by rocky** [comment]
I actually did more along these lines and have this around on my disk. However at some point things started to fail, and I hadn't been able to pinpoint why. 

So I'd like to stop this activity for now, see how this part goes, and then continue some other time.

File: test/test_convert.py (hunk 1 above)
Comment (mmatera) [reply test/test_convert.py:1]: stupid question: did you move it with `mv` or with `git mv`?

Comment (rocky) [reply test/test_convert.py:1]: Should be fixed now.

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**